pkg/manager/manager_test.go

```diff
@@ -1156,6 +1158,45 @@ var _ = Describe("manger.Manager", func() {
 			)
 		})
 
+		Context("with custom object selector", func() {
```

Comment:
I don't understand what exactly this is supposed to test and how this relates to the problem you described in the PR body? It also passes without your codechanges btw. It might be easier to unittest `combineScheme` directly

Reply:
I am trying to test the scenario mentioned above in the description. 

The main issue is when a scheme is combined using the BuilderWithOptions function, a new scheme object pointer is created and that pointer is configured into the cache - https://github.com/kubernetes-sigs/controller-runtime/blob/c3c1f058a9a080581e8fe99c004fcc792b2aff07/pkg/cache/cache.go#L300. 

The issue is that the original scheme object pointer is configured into the manager/controller and not the newly created one by combineScheme, so this means all the schemes added to `mgr.GetScheme()`, object are no available in the cache. As a result, the controller will fail to start, returning the error `"msg"="kind must be registered to the Scheme"`.

I think just unit-testing the `combineScheme` doesn't cover this scenario end to end. Also the `combineScheme` has unittest which already covers this entirely.